test/exe/main_common_test.cc

```diff
@@ -309,11 +309,12 @@ TEST_P(AdminRequestTest, AdminRequestAfterRun) {
 TEST_P(MainCommonTest, ConstructDestructLogger) {
   VERBOSE_EXPECT_NO_THROW(MainCommon main_common(argc(), argv()));
 
+  // log_msg constructor doesn't initialize all fields, resulting in uninitialized
+  // memory accesses.  "static" is a simple way to start from zeroed memory before
+  // the constructor is run, avoiding this problem.
+  // See https://github.com/gabime/spdlog/issues/888.
   const std::string logger_name = "logger";
-  spdlog::details::log_msg log_msg;
-  log_msg.logger_name = &logger_name;
-  log_msg.level = spdlog::level::level_enum::err;
-
+  static spdlog::details::log_msg log_msg(&logger_name, spdlog::level::level_enum::err);
```

Comment:
please comment this.

also, should we be concerned about the constructor in spdlog not initing what it should? Should we just roll back to a working version of spdlog until that's resolved?

Reply:
Josh, I'm confused.  What do you want commented beyond the comment two lines above?

I believe the default constructor has been removed from the upstream repo, so we could roll forward if we wanted.  Though the non-default constructor doesn't initialize all fields either.

I don't know how spdlog is supposed to be used, so I'm not sure if it's reasonable to demand they initialize all variables.  Maybe you could comment on the upstream issue?

Reply:
Here I think we should comment that we are declaring this static in order to zero-initialize it (which seems non-obvious to me), because it lacks a constructor. And point to the upstream issue.

Reply:
Sorry -- the comment is already there. Never mind :)